### PR TITLE
I1673 Add fixed codes; add 'static_cast's

### LINKED ISSUE #1673: Format string and type casting overflows
Running cppcheck against the Open GEE codebase reveals several possible format string and type casting overflows that we should fix, as listed below. This was run against 5.3.4-RC3.

```
%ld in format string (no. 1) requires 'long' but the argument type is 'unsigned long'. (invalidPrintfArgType_sint)
  src/common/packet.cpp, line 454, column 5

%ld in format string (no. 2) requires 'long' but the argument type is 'unsigned long'. (invalidPrintfArgType_sint)
  src/common/packet.cpp, line 454, column 5

When saving fgetc() return value in char variable there is loss of precision.  When fgetc() returns EOF this value is truncated. Comparing the char variable with EOF can have unexpected results. For instance a loop "while (EOF != (c = fgetc());" loops forever on some compilers/platforms and on other compilers/platforms it will stop when the file contains a matching character. (checkCastIntToCharAndBack)
  src/common/procpidstats.cpp, line 197, column 18

%lld in format string (no. 1) requires 'long long' but the argument type is 'unsigned long long'. (invalidPrintfArgType_sint)
  src/fusion/gst/gstValue.cpp, line 777, column 7

%d in format string (no. 13) requires 'int' but the argument type is 'unsigned int'. (invalidPrintfArgType_sint)
  src/fusion/mtjtools/scroll/imageview.cpp, line 1059, column 5

%d in format string (no. 14) requires 'int' but the argument type is 'unsigned int'. (invalidPrintfArgType_sint)
  src/fusion/mtjtools/scroll/imageview.cpp, line 1059, column 5

%d in format string (no. 15) requires 'int' but the argument type is 'unsigned int'. (invalidPrintfArgType_sint)
  src/fusion/mtjtools/scroll/imageview.cpp, line 1059, column 5

%d in format string (no. 16) requires 'int' but the argument type is 'unsigned int'. (invalidPrintfArgType_sint)
  src/fusion/mtjtools/scroll/imageview.cpp, line 1059, column 5

%d in format string (no. 17) requires 'int' but the argument type is 'unsigned int'. (invalidPrintfArgType_sint)
  src/fusion/mtjtools/scroll/imageview.cpp, line 1059, column 5

%d in format string (no. 18) requires 'int' but the argument type is 'unsigned int'. (invalidPrintfArgType_sint)
  src/fusion/mtjtools/scroll/imageview.cpp, line 1059, column 5

%x in format string (no. 3) requires 'unsigned int' but the argument type is 'unsigned char *'. (invalidPrintfArgType_uint)
  src/fusion/mtjtools/scroll/zoom.h, line 394, column 5

%x in format string (no. 3) requires 'unsigned int' but the argument type is 'Image'. (invalidPrintfArgType_uint)
  src/fusion/mtjtools/scroll/zoom.h, line 420, column 5

%x in format string (no. 1) requires 'unsigned int' but the argument type is 'void *'. (invalidPrintfArgType_uint)
  src/fusion/mtjtools/scroll/zoom.h, line 548, column 7

%d in format string (no. 2) requires 'int' but the argument type is 'size_t {aka unsigned long}'. (invalidPrintfArgType_sint)
  src/fusion/mtjtools/scroll/zoom.h, line 548, column 7

%x in format string (no. 1) requires 'unsigned int' but the argument type is 'void *'. (invalidPrintfArgType_uint)
  src/fusion/mtjtools/scroll/zoom.h, line 557, column 7

%d in format string (no. 2) requires 'int' but the argument type is 'size_t {aka unsigned long}'. (invalidPrintfArgType_sint)
  src/fusion/mtjtools/scroll/zoom.h, line 557, column 7

%d in format string (no. 3) requires 'int' but the argument type is 'size_t {aka unsigned long}'. (invalidPrintfArgType_sint)
  src/fusion/mtjtools/scroll/zoom.h, line 557, column 7

%d in format string (no. 4) requires 'int' but the argument type is 'size_t {aka unsigned long}'. (invalidPrintfArgType_sint)
  src/fusion/mtjtools/scroll/zoom.h, line 557, column 7

%x in format string (no. 1) requires 'unsigned int' but the argument type is 'void *'. (invalidPrintfArgType_uint)
  src/fusion/mtjtools/scroll/zoom.h, line 564, column 7

%u in format string (no. 1) requires 'unsigned int *' but the argument type is 'signed int *'. (invalidScanfArgType_int)
  src/fusion/portableglobe/geglxinfo.cpp, line 833, column 9

%u in format string (no. 1) requires 'unsigned int *' but the argument type is 'signed int *'. (invalidScanfArgType_int)
  src/fusion/portableglobe/geglxinfo.cpp, line 876, column 9

%d in format string (no. 1) requires 'int' but the argument type is 'ssize_t {aka signed long}'. (invalidPrintfArgType_sint)
  src/fusion/tools/makeprj.cpp, line 154, column 3

%lu in format string (no. 1) requires 'unsigned long' but the argument type is 'std::size_t {aka unsigned long}'. (invalidPrintfArgType_uint)
  src/server/mod_fdb/fusiondbservice.cpp, line 313, column 3

%lu in format string (no. 1) requires 'unsigned long' but the argument type is 'std::size_t {aka unsigned long}'. (invalidPrintfArgType_uint)
  src/server/mod_fdb/portableservice.cpp, line 457, column 3
```

### PR DESCRIPTION
**Pull request by tst-eclamar**
Add static casts and fix string format issues.
Fixes #1673

### CODE REVIEW

**Comment by tst-nfarah** [comment]
Do NOT merge yet until we create 5.3.4.1 branch.

**Comment by tst-mbiancalana** [comment]
5.3.4.1 branch was created, and I retargeted this PR. We're all set to proceed!

File: earth_enterprise/src/fusion/mtjtools/scroll/zoom.h [hunk 1]
```diff
@@ -417,7 +417,7 @@ class Image
   {
 #ifdef MALLOG
     log = fopen("c:/mallog.txt", "a"); // XXX
-    fprintf(log, "\nImage::Image(%d, %d, 0x%08x ... zoom)\n\n", x, y, source);
+    fprintf(log, "\nImage::Image(%d, %d, 0x%08x ... zoom)\n\n", x, y, static_cast<unsigned int>(source));
```
Comment (tst-lsavoie) [inline earth_enterprise/src/fusion/mtjtools/scroll/zoom.h:420]: This one confuses me. How are we able to cast an `Image &` to an `unsigned int`?

File: earth_enterprise/src/fusion/mtjtools/scroll/zoom.h [hunk 2]
```diff
@@ -545,7 +545,7 @@ class Image
     void *item = ::malloc(size);
 
     if (log != nullptr)
-      fprintf(log, "A 0x%08x %8d\n", item, size);
+      fprintf(log, "A 0x%08x %8lu\n", static_cast<unsigned int>(item), static_cast<unsigned long>(size));
```
Comment (tst-lsavoie) [inline earth_enterprise/src/fusion/mtjtools/scroll/zoom.h:548]: Minor: there is a `%p` printf specifier for `void *` that would make more sense for `item`. The same applies to the two changes below.

File: earth_enterprise/src/fusion/mtjtools/scroll/zoom.h (hunk 1 above)
Comment (tst-lsavoie) [reply earth_enterprise/src/fusion/mtjtools/scroll/zoom.h:420]: I did some research on this and there are some compile failures if you `#define MALLOG` at the top of this file. I'd be fine with simply removing all the MALLOG sections from this file. That would fix the issue and we won't loose anything we're using or we can't recreate if we need to.

**Pull request merged**